PusherClient/Pusher.cs

```diff
@@ -133,6 +133,9 @@ public void Connect()
 
             UnregisterEventsOnConnection();
 
+            if (_connection != null)
+                _connection.Disconnect();
+

```

Comment:
With my PR, isn't this case impossible? i.e. there should only ever be a single Connection instantiated and assigned to _connection.